### DISCUSSION TITLE
where is Third-Party Source Code option

### DISCUSSION THREAD

**Discussion by Pandai20** [zeroc-ice/ice#1403] [Q&A] [answered]
I couldn't find the Third-Party Source Code option on zeroc，would you please help me？/(ㄒoㄒ)/~~

**Comment by pepone** [comment]
Hey,

Can you clarify what are you looking for? on most systems, all the third-party dependencies are provided by the OS, for Windows we provide NuGet packages.

**Comment by Pandai20** [comment]
My build environment is cnetos7 and I followed some suggestions and it lets me install mcpp, bzip2 etc dependencies before compiling. Then it tells me that there is this installation package on the official zeroc website. But I looked for a long time and couldn't find it, so do you mean I don't have to install these, just make it?  thanks(*^_^*)

**Comment by pepone** [comment, marked as answer]
If you want to build Ice from sources you need to follow instructions from https://github.com/zeroc-ice/ice/tree/3.7/cpp

For Centos 7 it is simpler to just use our Red Hat 7 packages, see https://zeroc.com/downloads/ice/3.7/cpp#redhat

**Comment by Pandai20** [comment]
Thank you, I finally 'make' successfully. Because I installed the bzip,epel,Berkeley DB. And put the path to in the Make.rules.It took me about two or three days. Thank you for reminding me
